lighthouse-core/gather/driver/prepare.js

```diff
@@ -0,0 +1,177 @@
+ * @param {Pick<LH.Config.NavigationDefn, 'disableThrottling'|'disableStorageReset'|'blockedUrlPatterns'> & {url: string}} navigation
+ */
+async function prepareNetworkForNavigation(session, settings, navigation) {
+  const status = {msg: 'Preparing network conditions', id: `lh:gather:prepareNetworkForNavigation`};
+  log.time(status);
+
+  if (navigation.disableThrottling) await emulation.clearThrottling(session);
+  else await emulation.throttle(session, settings);
+
+  // Set request blocking before any network activity.
+  // No "clearing" is done at the end of the navigation since Network.setBlockedURLs([]) will unset all if
+  // neccessary at the beginning of the next navigation.
+  const blockedUrls = (navigation.blockedUrlPatterns || []).concat(
+    settings.blockedUrlPatterns || []
+  );
+  await session.sendCommand('Network.setBlockedURLs', {urls: blockedUrls});
+
+  const headers = settings.extraHeaders;
+  if (headers) await session.sendCommand('Network.setExtraHTTPHeaders', {headers});
+
+  log.timeEnd(status);
+}
+
+/**
+ * Prepares a target to be analyzed in navigation mode by enabling protocol domains, emulation, and new document
+ * handlers for global APIs or error handling.
+ *
+ * This method should be used in combination with `prepareTargetForIndividualNavigation` before a specific navigation occurs.
+ *
+ * @param {LH.Gatherer.FRTransitionalDriver} driver
+ * @param {LH.Config.Settings} settings
+ */
+async function prepareTargetForNavigationMode(driver, settings) {
+  // Emulate our target device screen and user agent.
+  await emulation.emulate(driver.defaultSession, settings);
+
+  // Enable better stacks on network requests.
+  await enableAsyncStacks(driver.defaultSession);
+
+  // Automatically handle any JavaScript dialogs to prevent a hung renderer.
+  await dismissJavaScriptDialogs(driver.defaultSession);
+
+  // Inject our snippet to cache important web platform APIs before they're (possibly) ponyfilled by the page.
+  await driver.executionContext.cacheNativesOnNewDocument();
+
+  // Wrap requestIdleCallback so pages under simulation receive the correct rIC deadlines.
+  if (settings.throttlingMethod === 'simulate') {
+    await shimRequestIdleCallbackOnNewDocument(driver, settings);
+  }
+}
+
+/**
+ * Prepares a target for a particular navigation by resetting storage and setting network.
+ *
+ * This method assumes `prepareTargetForNavigationMode` has already been invoked.
+ *
+ * @param {LH.Gatherer.FRProtocolSession} session
+ * @param {LH.Config.Settings} settings
+ * @param {Pick<LH.Config.NavigationDefn, 'disableThrottling'|'disableStorageReset'|'blockedUrlPatterns'> & {url: string}} navigation
+ * @return {Promise<{warnings: Array<LH.IcuMessage>}>}
+ */
+async function prepareTargetForIndividualNavigation(session, settings, navigation) {
+  /** @type {Array<LH.IcuMessage>} */
+  const warnings = [];
+
+  const shouldResetStorage = !settings.disableStorageReset && !navigation.disableStorageReset;
```

Comment:
storage reset was moved here for three main reasons:

1. It requires a URL, so it's already specific to a navigation intent, doesn't make sense as a general target setup.
2. If we're trying to clear storage, that we should clear before *each* navigation. We've come a long way in 5 years of Lighthouse and have clarity that `passes`/`navigations` are *not* going to be how we solve flows, we're solving flows with FR, so additional navigations should be reloading the same situation unless configured otherwise :)
3. It does not change the default config sequence anyway, only if you had custom config combinations of `recordTrace: false, useThrottling: true` will it matter.